packages/core/src/components/Modal/ModalFooter/ModalFooterButtons/__tests__/modalFooterButtons-tests.jest.tsx

```diff
@@ -42,5 +42,20 @@ describe("ModalFooterButtons", () => {
       expect(onSecondaryButtonClick).toBeCalledTimes(1);
       expect(onPrimaryButtonClick).not.toBeCalled();
     });
+
+    it("should not call onPrimaryButtonClick after click confirm when primary button is disabled", () => {
+      const onPrimaryButtonClick = jest.fn();
+      const onSecondaryButtonClick = jest.fn();
+      const modalFooterComponent = renderComponent({
+        primaryButtonText: PRIMARY_BUTTON_TEXT,
+        secondaryButtonText: SECONDARY_BUTTON_TEXT,
+        disablePrimaryButton: true,
+        onPrimaryButtonClick,
+        onSecondaryButtonClick
+      });
+      fireEvent.click(modalFooterComponent.getByText(PRIMARY_BUTTON_TEXT));
+      expect(onPrimaryButtonClick).not.toBeCalled();
+      expect(onSecondaryButtonClick).not.toBeCalled();
```

Comment:
This is redundant no?

Reply:
this is a test to validate that the button is indeed disabled

Reply:
I mean just this line, there's nothing going on with the secondary button in this test, no?